gym/envs/registration.py

```diff
@@ -621,30 +621,48 @@ def make(
     mode = _kwargs.get("render_mode")
     apply_human_rendering = False
 
-    # If we have access to metadata we check that "render_mode" is valid
-    if hasattr(env_creator, "metadata"):
```

Comment:
Could we just do 

```python
if hasattr(env_creator, "metadata") and "render_modes" in env_creator.metadata:
    ...
elif hasattr(env_creator, "metadata") and "render_modes"  not in env_creator.metadata:  # Longer for readability
    logger.warn("Metadata doesn't specify 'render_modes'.")
```

That would probably avoid all of the nested if statements and cover what's relevant imo (check the other two comments).